src/cr/cube/crunch_cube.py

```diff
@@ -752,11 +777,21 @@ def labels(self, include_missing=False, include_transforms_for_dims=False):
         Returns
             labels (list of lists): Labels for each dimension
         '''
-        return [
+        labels = [
             dim.labels(include_missing, include_transforms_for_dims)
             for dim in self.dimensions
         ]
 
+        if prune:
```

Comment:
Could you extract all that's under this `if` in a separate method, and possibly call it `_prune_labels`?